tools/const_finder/lib/const_finder.dart

```diff
@@ -53,15 +51,22 @@ class _ConstVisitor extends RecursiveVisitor<void> {
 
   @override
   void visitInstanceConstantReference(InstanceConstant node) {
+    node.fieldValues.values.whereType<InstanceConstant>().forEach(visitInstanceConstantReference);
     if (!_matches(node.classNode)) {
+      super.visitInstanceConstantReference(node);
       return;
     }
     final Map<String, dynamic> instance = <String, dynamic>{};
     for (MapEntry<Reference, Constant> kvp in node.fieldValues.entries) {
+      if (kvp.value is! PrimitiveConstant<dynamic>) {
+        continue;
+      }
       final PrimitiveConstant<dynamic> value = kvp.value as PrimitiveConstant<dynamic>;
       instance[kvp.key.canonicalName.name] = value.value;
     }
-    constantInstances.add(instance);
+    if (_visitedInstnaces.add(instance.toString())) {
```

Comment:
This is cheating a little, but it's simpler than creating a class that can properly hash and equality compare a dynamic map.

It's good enough to dedup things when all the parameters are in order.